doc/contributing/CODING_STANDARDS.md

```diff
@@ -176,6 +176,16 @@ when binding a C++ `enum class` using pybind, use
 namespaced in the same way as in C++, i.e. not polluting the parent
 namespace.
 
+### Const
+
+Methods of interface classes that are conceptually stateless, and their
+corresponding wrapper class methods should not be marked `const`. For
+example, `ManagerInterface` + `Manager`, `HostInterface` and `Host`.
+This may seem to contradict the "stateless" definition, but it allows
+the implementation to maintain private state to service the API requests
+efficiently, such as server connections or caches. Exceptions to this
+are the identification methods that should be compile-time constant.
```

Comment:
[consider] "should be compile-time constant" might be read more literally than intended. Perhaps
```suggestion
are the identification methods that should be (effectively) compile-time constant.
```
or maybe there's a better qualification

Reply:
Heck we should make them `constexpr` if that were the case. (Although i doubt we can for most if not all, due to constexpr not being available on virtual methods pre c++20.)
I think keeping this simple may be best, don't bring up the messy idea of compile-time constness at all, simply:
```
Exceptions to this are the identification methods that should be constants.
```

Reply:
Updated - (rebased and) force-pushed.